Tests/ServiceDiscoveryTests/Utility.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftServiceDiscovery open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftServiceDiscovery project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftServiceDiscovery project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Dispatch
+
+@testable import ServiceDiscovery
+import XCTest
+
+extension XCTestCase {
+    func ensureResult<SD: ServiceDiscovery>(serviceDiscovery: SD, service: SD.Service) throws -> Result<[SD.Instance], Error> {
+        let semaphore = DispatchSemaphore(value: 0)
+        var result: Result<[SD.Instance], Error>?
+
+        serviceDiscovery.lookup(service, deadline: nil) {
+            result = $0
+            semaphore.signal()
+        }
+
+        _ = semaphore.wait(timeout: DispatchTime.now() + .seconds(1))
+
+        guard let _result = result else {
+            throw LookupError.timedOut
+        }
+
+        return _result
+    }
+}
+
+#if compiler(>=5.5)
+extension XCTestCase {
+    @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+    func runAsyncAndWaitFor(_ closure: @escaping (_ finished: XCTestExpectation) async -> Void, _ timeout: TimeInterval = 1.0) {
+        let finished = expectation(description: "finished")
+        detach {
+            await closure(finished)
+        }
+        wait(for: [finished], timeout: timeout)
+    }
```

Comment:
Could this not be:

```
        let finished = expectation(description: "finished")
         detach {
             await closure()
             finished.fulfill() // maybe a do catch if necessary
         }
         wait(for: [finished], timeout: timeout)
     }
```

this way the tests don't have to call the fulfill -- it's more similar to what we used to have in the impl.

---

add a TODO to remove this once XCTest supports async functions?

Reply:
amended https://github.com/apple/swift-service-discovery/pull/35/commits/e48b5fe766e7f706ae6e558960fc4ade22c4f5e6